flash attn pip install

mhenrichsen opened the pull request:
Untested.

Should pip install flash attn 2.0.8 instead of building wheels.

Fixes: https://github.com/OpenAccess-AI-Collective/axolotl/issues/414

Commit:
flash attn pip

Commit:
add packaging

Commit:
add packaging to apt get

mhenrichsen commented:
I can't figure out why it can't find `packaging`...

Commit:
install flash attn in dockerfile

winglian commented on docker/Dockerfile-base:
```diff
@@ -90,13 +72,8 @@ RUN mkdir -p /workspace/wheels/bitsandbytes
-COPY --from=flash-attn-builder /workspace/flash-attention/csrc/layer_norm/dist/dropout_layer_norm-*.whl wheels
 
-RUN pip3 install wheels/deepspeed-*.whl wheels/flash_attn-*.whl wheels/fused_dense_lib-*.whl wheels/xentropy_cuda_lib-*.whl wheels/rotary_emb-*.whl wheels/dropout_layer_norm-*.whl
+RUN pip3 install wheels/deepspeed-*.whl wheels/fused_dense_lib-*.whl wheels/xentropy_cuda_lib-*.whl wheels/rotary_emb-*.whl wheels/dropout_layer_norm-*.whl
```
Don't forget to remove the other wheels that flash attn builds.

mhenrichsen replied:
These?
`wheels/fused_dense_lib-*.whl wheels/xentropy_cuda_lib-*.whl wheels/rotary_emb-*.whl wheels/dropout_layer_norm-*.whl`

mhenrichsen replied:
Should be good now.

winglian reviewed (changes requested):
Also need to add flash-attn to the requirements.txt file. Thanks!

mhenrichsen commented:
> Also need to add flash-attn to the requirements.txt file. Thanks!

That's what I did to begin with, but it fails because it can't find `packaging`.

Commit:
remove unused whls

Commit:
add wheel

winglian committed:
clean up pr

fix packaging requirement for ci
upgrade pip for ci
skip build isolation for requiremnents to get flash-attn working
install flash-attn seperately

winglian committed:
install wheel for ci

winglian committed:
no flash-attn for basic cicd

winglian committed:
install flash-attn as pip extras

winglian committed:
Merge branch 'main' into chore/414-flash-attn-wheels
